lib/puppet/util/windows/user.rb

```diff
@@ -52,19 +60,18 @@ def check_token_membership
   module_function :check_token_membership
 
   def password_is?(name, password)
-    # Windows API LogonUserW always disallows this with error 1327:
-    # Logon failure: user account restriction. Possible reasons are blank passwords
-    # not allowed, logon hour restrictions, or a policy restriction has been enforced.
-    if password.nil? || password == ''
-      raise Puppet::Util::Windows::Error.new(
```

Comment:
Per commit msg in f45da40d I found we can successfully supply both `nil` and `''` (zero-length) passwords -  strictly speaking, from an external interface perspective, I think all that's needed is to remove the second operand to `||` (check for `''`) but I also didn't still see a reason to `raise` on `nil`. That being said, `LogonUserW` doesn't appear to distinguish between a password of value `NULL` and of value `''`, so perhaps that would be a reason to `raise` on `nil` - prevent ambiguity? Not sure..